docs/platform-rpi3.md

```diff
@@ -0,0 +1,63 @@
+# LinuxKit with Raspberry Pi 3
+
+LinuxKit arm64 images can be booted on a Raspberry Pi Model 3b. We
+strongly recommend using a serial console for testing and development,
+though the normal video console with a USB keyboard should also work.
+
+The standard LinuxKit arm64 kernels do not support certain features of
+the Raspberry Pi 3b, in particular:
+
+- Wireless networking
+- Bluetooth
+- Graphics
+
+The main reason is that we do not want to add the relevant drivers to
+the default kernel configuration as the default LinuxKit kernels are
+mostly targeted at VMs and bare-metal servers.
+
+**We do not support any 32bit only Raspberry Pi models such as the Pi zero**
+
+## Boot
+
+**You currently need to build the Raspberry Pi images on a arm64 linux
+machine and we only support booting 4.9.x or newer kernels.**
+
+
+The `moby` tool has a special output format for the Raspberry, `rpi3`,
+when run on a `arm64` machine:
+
+```
+moby build -format rpi3 foo.yml
+```
+
+The output is a tar-ball which can be extracted into the root
+directory of a FAT32 formatted SD card. Insert the SD card into the
+slot on the Raspberry Pi and power it on.
+
+The boot uses [`u-boot`](https://www.denx.de/wiki/U-Boot). `u-boot` is
+used to load the kernel, initial RAM disk, and the device tree
+file. The Raspberry Pi is booted and run from the ramdisk.
+
+TODO: Document network booting.
+
+
+## Console
+
+Both the serial console, via a suitable USB adapter, and the video
+console with a USB keyboard should work. As mentioned above, there is
+not graphics support.
+
+
+## Disks
+
+The SD card is used to boot. It should be mountable as a disk in
+LinuxKit but is not really suitable for persistent storage.
+
+TODO: Work out how to use a second partition on the SD card for storage.
```

Comment:
For this part I'd been playing with `sfdisk` a little as in https://github.com/hypriot/image-builder-raw/blob/master/builder/rpi/build.sh#L33..L40

I couldn't get this to fully work, mind you, and I didn't hack on this for long. But it works for HypriotOS, so should be viable.